pkg/deploy/defaults.go

```diff
@@ -360,7 +360,7 @@ func DefaultPullPolicyFromDockerImage(dockerImage string) string {
 }
 
 func GetSingleHostExposureType(cr *orgv1.CheCluster) string {
-	if util.IsOpenShift {
+	if util.IsOpenShift || cr.Spec.DevWorkspace.Enable {
```

Comment:
Why do we need this changes? 
So, it means that single-host will be default for k8s when devworkspace is enabled.

Reply:
Exactly. There was the agreement on the last cabal that we're doing that, if I remember correctly.

Reply:
Then `DefaultOpenShiftSingleHostExposureType` const has no sense anymore.
Let's remove it and just return `gateway`

Reply:
there will be only `gateway` with devworkspaces. `native` does not really work with native auth. Everything we want to protect with oauth-proxy and kube-rbac-proxy has to go through the gateway.